internal/resolution/variablesources/installed_package.go

```diff
@@ -0,0 +1,83 @@
+package variablesources
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/operator-framework/deppy/pkg/deppy"
+	"github.com/operator-framework/deppy/pkg/deppy/input"
+
+	olmentity "github.com/operator-framework/operator-controller/internal/resolution/entities"
+	"github.com/operator-framework/operator-controller/internal/resolution/util/predicates"
+	"github.com/operator-framework/operator-controller/internal/resolution/util/sort"
+	"github.com/operator-framework/operator-controller/internal/resolution/variables"
+)
+
+var _ input.VariableSource = &InstalledPackageVariableSource{}
+
+type InstalledPackageVariableSource struct {
+	bundleImage string
+}
+
+func (r *InstalledPackageVariableSource) GetVariables(ctx context.Context, entitySource input.EntitySource) ([]deppy.Variable, error) {
+	// find corresponding bundle entity for the installed content
+	resultSet, err := entitySource.Filter(ctx, predicates.WithBundleImage(r.bundleImage))
+	if err != nil {
+		return nil, err
+	}
+	if len(resultSet) == 0 {
+		return nil, r.notFoundError()
+	}
+
+	// sort by channel and version
+	// TODO: this is a bit of a hack and it assumes a well formed catalog.
```

Comment:
could you please create issue/s for these  todos and link them in the comments here?